### PR TITLE
:warning: Fix enableBMHNameBasedPreallocation label to ENABLE_BMH_NAME_BASED_PR…

### PR DESCRIPTION
**Pull request by peppi-lotta**
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #https://github.com/metal3-io/cluster-api-provider-metal3/issues/1242

### CODE REVIEW

**Review by smoshiur1237 — COMMENTED** [review]
/approve

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ spec:
             - /manager
           args:
             - "--webhook-port=9443"
-            - "--enableBMHNameBasedPreallocation=${enableBMHNameBasedPreallocation:=false}"
+            - "--enableBMHNameBasedPreallocation=${ENABLE_BMH_NAME_BASED_PREALLOCATION:=false}"
```
Comment (tuminoid) [inline config/manager/manager.yaml:25]: Q: Where is this env name coming? It is not present anywhere in Metal3 right now.

Comment (tuminoid) [reply config/manager/manager.yaml:25]: OK, the issue links into discussion and it leads to https://github.com/kubernetes-sigs/cluster-api/pull/9513/files#diff-4127eae8106dc3a0709a4131a0007cc3dd8b0393ce07916829b1b4394603aecfR62 which seems also relevant as it says the upper case should be also used in clusterctl config. `scripts/ci-e2e.sh` L40 keeps the lowercase format though, should that also be fixed?

/cc @kashifest

**Comment by kashifest** [comment]
/hold
For the change requested and feature test to pass

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Comment by tuminoid** [comment]
> /hold For the change requested and feature test to pass

All tests are passed, shall we unhold?

**Comment by Rozzii** [comment]
I will check what docs will need to change to support this then we can merge.

**Comment by Rozzii** [comment]
@adilGhaffarDev has volunteered to create an updated version of this PR and we will try to get this merged.

**Comment by adilGhaffarDev** [comment]
I have rebased it, please check.

**Comment by adilGhaffarDev** [comment]
LGTM from my side @mboukhalfa @Rozzii please check

**Review by Rozzii — COMMENTED** [review]
/lgtm

**Comment by Rozzii** [comment]
/hold cancel
let's see what breaks :D

**Pull request merged**